packages/zoe/src/contracts/multipoolAutoswap/constantProduct.js

```diff
@@ -0,0 +1,38 @@
+// @ts-check
+
+import { assert, details as X } from '@agoric/assert';
+import { natSafeMath } from '../../contractSupport';
+
+// A pool seat has Central and Secondary keywords, and a swap seat has
+// In and Out keywords
+const isPoolSeat = allocation => {
+  return allocation.Central !== undefined || allocation.Secondary !== undefined;
+};
+
+const calcK = allocation => {
+  return natSafeMath.multiply(
+    allocation.Secondary.value,
+    allocation.Central.value,
+  );
+};
+
+/**
+ *
+ * @param {SeatStaging[]} stagings
+ */
+export const assertConstantProduct = stagings => {
+  stagings.forEach(seatStaging => {
```

Comment:
Once we have a few examples, I can easily imagine wanting a "iterate the pairs of old and new allocations".

Reply:
Yup, we can do that now for a different assert function. Each seat staging has the ability to get the prior allocation and the proposed allocation.